packages/@react-stately/data/src/useTreeData.ts

```diff
@@ -106,6 +111,11 @@ export interface TreeData<T extends object> {
    */
   move(key: Key, toParentKey: Key, index: number): void,
 
+  // TODO for a moveBefore and moveAfter
+  // Case 1: you are moving items of the same parent within the same parent
+  // Case 2: you are moving items from a different parent
+  // Case 3: you are moving a mix of items
+  // Alternatively
```

Comment:
Another TODO for this PR: will want something like moveBefore/moveAfter like we have for useListData for reorder operations. One alternative would be to just to dupe the items to move + modify their keys -> insert them accordingly -> delete the original -> revert the keys of the items that were moved